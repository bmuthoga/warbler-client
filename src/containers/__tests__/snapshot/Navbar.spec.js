import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Navbar } from '../../Navbar'

const props = {
  logout: jest.fn(),
  currentUser: {
    isAuthenticated: false,
    user: {}
  }
}

describe('Navbar Snapshot Tests', () => {
  it('should render the signed out state Navbar component', () => {
    const wrapper = shallow(
        <Navbar { ...props } />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
  it('should render the signed in state Navbar component', () => {
    const currentUser = {
      isAuthenticated: true,
      user: {
        id: '848hrjhgj48u4585',
        username: 'testuser',
        profileImageUrl: 'http://lorempixel.com/200/200/'
      }
    }
    const wrapper = shallow(
      <Navbar { ...props } currentUser={currentUser} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
