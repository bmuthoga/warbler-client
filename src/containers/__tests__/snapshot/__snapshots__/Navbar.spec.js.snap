// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot Tests should render the signed in state Navbar component 1`] = `
<nav
  className="navbar navbar-expand"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <Link
        className="navbar-brand"
        to="/"
      >
        <img
          alt="Warbler Home"
          src="warbler-logo.png"
        />
      </Link>
    </div>
    <ul
      className="nav navbar-nav navbar-right"
    >
      <li>
        <Link
          to="/users/848hrjhgj48u4585/messages/new"
        >
          New Message
        </Link>
      </li>
      <li>
        <a
          onClick={[Function]}
        >
          Log out
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navbar Snapshot Tests should render the signed out state Navbar component 1`] = `
<nav
  className="navbar navbar-expand"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <Link
        className="navbar-brand"
        to="/"
      >
        <img
          alt="Warbler Home"
          src="warbler-logo.png"
        />
      </Link>
    </div>
    <ul
      className="nav navbar-nav navbar-right"
    >
      <li>
        <Link
          to="/signup"
        >
          Sign up
        </Link>
      </li>
      <li>
        <Link
          to="/signin"
        >
          Log in
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
